src/Units/UnitModel.java

```diff
@@ -216,4 +216,24 @@ public boolean hasActionType(ActionType actionType)
     }
     return hasAction;
   }
+
+  public boolean isSurfaceUnit()
+  {
+    return (ChassisEnum.SHIP == chassis) || (ChassisEnum.TANK == chassis) || (ChassisEnum.TROOP == chassis);
+  }
```

Comment:
Hmm. I was kind of planning to use that for determining whether terrain defense applies... I'd forgotten we have a separate chassis for submerged units.
I'm not sure what to say about that.

Reply:
Could just do `if( !unit.model.isAirUnit() )`. Not quite as idiomatic as you probably wanted though, but adding `isAffectedByTerrain()` feels a little obtuse.